packages/algorand/src/htlc/index.ts

```diff
@@ -0,0 +1,149 @@
+import Config from '../config';
+import { AlgorandProvider, AlgorandWallet } from '@jelly-swap/types';
+import { fundHTLCContract, formatNote} from './utils';
+import { fixHash, sha256, generateHashLock } from '@jelly-swap/utils';
+import { RefundEvent } from '../types';
+import {HttpClient} from '../providers';
+import algosdk from 'algosdk';
+
+export default class HTLC {
+    provider: AlgorandProvider;
+    wallet: AlgorandWallet;
+    htlcTemplate = require('algosdk/src/logicTemplates/htlc');
+    config: any;
+
+    constructor(wallet: any, config = Config()) {
+        this.config = config;
+        this.wallet = wallet;
+        this.provider = wallet.provider;
+    }
+
+    public async newSwap(
+        value: number,
+        recipientAddress: string,
+        refundAddress: string,
+        hashLock: string,
+        expiration: number = this.config.expiration,
+        metadata: any
+    ) {
+        try {
+            const params = await this.provider.getTransactionParams();
+            const hashFn = 'sha256';
+            const htlc = new this.htlcTemplate.HTLC(refundAddress, recipientAddress, hashFn, hashLock, expiration, this.config.maxFee);
+
+            return fundHTLCContract(params, htlc, this.wallet, value, this.provider, metadata);
+        } catch (error) {
+            console.log(error);
+        }
+    }
+
+    public async withdraw(
+        initiationTxHash: string,
+        recipientAddress: string,
+        refundAddress: string,
+        expiration: number,
+        secret: string,
+        metadata: any,
+        secretHash?: any
+    ) {
+        try {
+            const params = await this.provider.getTransactionParams();
+            const hashFn = 'sha256';
+            if (!secretHash) {
+                secretHash = sha256(secret);
+            }
+
+            const htlc = new this.htlcTemplate.HTLC(
+                refundAddress,
+                recipientAddress,
+                hashFn,
+                secretHash,
+                expiration,
+                this.config.maxFee
+            );
+
+            const txn = {
+                from: htlc.getAddress(),
+                to: Config().zeroAddress,
```

Comment:
Use `this.config`

Reply:
This is a constant, I would like to take it from the config

Reply:
Then remove `zeroAddress` from the config and put it in the file that is used instead.